core/trino-main/src/main/java/io/trino/operator/join/unspilled/JoinProbe.java

```diff
@@ -100,14 +122,27 @@ private static long[] fillCache(
             LookupSource lookupSource,
             Page page,
             Block probeHashBlock,
-            Page probePage)
+            Page probePage,
+            boolean isRle)
     {
         int positionCount = page.getPositionCount();
         List<Block> nullableBlocks = IntStream.range(0, probePage.getChannelCount())
                 .mapToObj(i -> probePage.getBlock(i))
                 .filter(Block::mayHaveNull)
                 .collect(toImmutableList());
 
+        if (isRle) {
+            long[] joinPositionCache = new long[1];
+            if (nullableBlocks.stream().anyMatch(block -> block.isNull(0))) {
```

Comment:
maybe add a comment here that if any probe value is null that it will not match with any build row